packages/fluentui/react-bindings/src/styles/resolveVariables.ts

```diff
@@ -10,33 +10,50 @@ import {
 const variablesCache = new WeakMap<ThemePrepared, Record<string, ComponentVariablesObject>>();
 
 const resolveVariables = (
-  displayName: string,
+  displayNames: string[],
   theme: ThemePrepared,
   variables: ComponentVariablesInput | undefined,
   enabledVariablesCaching: boolean | undefined
 ): ComponentVariablesObject => {
+  let componentThemeVariables: ComponentVariablesObject;
+
+  // Filter out components that doesn't have defined variables in theme
+  const effectiveDisplayNames = displayNames.filter(displayName => !!theme.componentVariables[displayName]);
+
   //
   // Simple caching model, works only if there is no `props.variables`
   // Resolves variables for this component, cache the result in provider
   //
 
-  let componentThemeVariables = {};
-
   if (enabledVariablesCaching) {
     if (!variablesCache.has(theme)) {
       variablesCache.set(theme, {});
     }
 
     const variablesThemeCache = variablesCache.get(theme) || {};
 
-    if (!variablesThemeCache[displayName]) {
-      variablesThemeCache[displayName] = callable(theme.componentVariables[displayName])(theme.siteVariables) || {};
+    // const displayNames = { Foo: variables, Bar: undefined, Baz: undefined }
+    // This allows to avoid creating useless cache entries for `Bar` & `Baz` components
+    const handlingDisplayName = effectiveDisplayNames[effectiveDisplayNames.length - 1];
+
+    if (!variablesThemeCache[handlingDisplayName]) {
+      effectiveDisplayNames.map(displayName => theme.componentVariables[displayName]);
+
+      variablesThemeCache[handlingDisplayName] = mergeComponentVariables(
+        ...effectiveDisplayNames.map(displayName => theme.componentVariables[displayName])
+      )(theme.siteVariables);
       variablesCache.set(theme, variablesThemeCache);
     }
 
-    componentThemeVariables = variablesThemeCache[displayName];
+    componentThemeVariables = variablesThemeCache[handlingDisplayName];
   } else {
-    componentThemeVariables = callable(theme.componentVariables[displayName])(theme.siteVariables) || {};
+    if (effectiveDisplayNames.length === 1) {
+      componentThemeVariables = callable(theme.componentVariables[effectiveDisplayNames[0]])(theme.siteVariables) || {};
+    } else {
+      componentThemeVariables = mergeComponentVariables(...effectiveDisplayNames.map(displayName => theme.componentVariables[displayName]))(
```

Comment:
We should be able to just take this from the cache. No need for merging again as far as I can see

Reply:
This brunch is out of caching: